File: Python/k07.py
from executable import executable

class k07(executable):

    def show_texts(self):
        print("Hello World!")
        print("ようこそ")
        print("Python言語の世界へ！")

    def get_max_value(self, value1, value2, value3):
        temp = value1 if value1 >= value2 else value2
        temp = temp if temp >= value3 else value3
        return temp

    def get_type(self, age):
        if age <= 0:
            return 0
        elif age < 3 or age >= 70:
            return 1
        elif age >= 3 and age <= 15:
            return 2
        elif age >= 60 and age < 70:
            return 3
        else:
            return 4

    def question1(self):
        choice = 1
        try:
            while choice != 0:
                print("メッセージを表示しますか？（０：終了する　１：表示する）＞", end="")
                choice = int(input())
                if choice == 1:
                    self.show_texts()
            print("終了します。")
        except ValueError as ve:
            print(ve)

    def question2(self):
        try:
            numbers = []
            for x in range(1, 4):
                print("{}つ目の値を入力してください。＞".format(x), end="")
                numbers.append(int(input()))

            max_value = self.get_max_value(numbers[0], numbers[1], numbers[2])
            print("{}つの中で最大値は{}です。".format(len(numbers), max_value))

        except ValueError as ve:
            print(ve)

    def question3(self):
        try:
            age = 0
            no = 0
            print("年齢を入力して下さい。＞", end="")
            age = int(input())

            no = self.get_type(age)
            dispatcher = {
                0: "不適切な値が入力されました。",
                1: "入場料金無料です。",
                2: "子供料金で半額です。",
                3: "シニア割引で１割引きです。"
                }

            print(dispatcher.get(no, "通常料金です。"))

        except ValueError as ve:
            print(ve)

File: rust/src/k07.rs
use crate::Executable;
use crate::read_user_input::read_user_input;
use std::io::Write;

pub struct K07 {}

impl K07 {
    pub fn new() -> K07 {
        K07 {}
    }

    fn show_text(&self) {
        println!("Hello World!");
        println!("ようこそ");
        println!("Rust言語の世界へ！");
    }

    fn get_max_value(&self, value1: i32, value2: i32, value3: i32) -> i32 {
        let mut temp = if value1 >= value2 {
            value1
        } else { value2 };
        temp = if temp >= value3 {
            temp
        } else {
            value3
        };
        temp
    }

    fn get_type(&self, age: i32) -> i32 {
        match age {
            x if x <= 0 => 0,
            x if x < 3 || age >= 70 => 1,
            x if x >= 3 && age <= 15 => 2,
            x if x >= 60 && age < 70 => 3,
            _ => 4
        }
    }
}

impl Executable for K07 {
    fn question_1(&self) {
        let mut choice = 1;
        while choice != 0 {
            print!("メッセージを表示しますか？（０：終了する　１：表示する）＞");
            std::io::stdout().flush().unwrap();
            let mut input_string = String::new();
            choice = match read_user_input(&mut input_string) {
                Ok(x) => x,
                Err(e) => {
                    println!("エラーが発生しました：{}", e.to_string());
                    return;
                }
            };
            if choice == 1 {
                self.show_text();
            }
        }
        println!("終了します。");
    }

    fn question_2(&self) {
        let mut numbers: Vec<i32> = Vec::new();
        for i in 1..4 {
            print!("{}つ目の値を入力してください。＞", i);
            std::io::stdout().flush().unwrap();
            let mut input_string: String = String::new();
            numbers.push(match read_user_input(&mut input_string) {
                Ok(x) => x,
                Err(e) => {
                    println!("エラーが発生しました：{}", e.to_string());
                    return;
                }
            })
        }

        let max_value = self.get_max_value(numbers[0], numbers[1], numbers[2]);
        println!("{}つの中で最大値は{}です。", numbers.len() as i32, max_value);
    }

    fn question_3(&self) {
        let no: i32;
        print!("年齢を入力して下さい。＞");
        std::io::stdout().flush().unwrap();
        let mut age = String::new();
        let age = match read_user_input(&mut age) {
            Ok(x) => x,
            Err(e) => {
                println!("エラーが発生しました：{}", e.to_string());
                return;
            }
        };

        no = self.get_type(age);
        match no {
            0 => println!("不適切な値が入力されました。"),
            1 => println!("入場料金無料です。"),
            2 => println!("子供料金で半額です。"),
            3 => println!("シニア割引で一割引きです。"),
            _ => println!("通常料金です。")
        }
    }

    fn question_4(&self) {
        return;
    }
}